frappe/utils/data.py

```diff
@@ -473,23 +473,23 @@ def get_weekday(datetime=None):
 def get_timespan_date_range(timespan):
 	today = nowdate()
 	date_range_map = {
-		"last week": lambda: (add_to_date(today, days=-7), today),
-		"last month": lambda: (add_to_date(today, months=-1), today),
-		"last quarter": lambda: (add_to_date(today, months=-3), today),
-		"last 6 months": lambda: (add_to_date(today, months=-6), today),
-		"last year": lambda: (add_to_date(today, years=-1), today),
+		"last week": lambda: (get_first_day_of_week(add_to_date(today, days=-7)), get_last_day_of_week(add_to_date(today, days=-7))),
+		"last month": lambda: (get_first_day(add_to_date(today, months=-1)), get_last_day(add_to_date(today, months=-1))),
+		"last quarter": lambda: (get_quarter_start(add_to_date(today, months=-3),as_str=True), get_quarter_ending(add_to_date(today, months=-3))),
+		"last 6 months": lambda: (get_quarter_start(add_to_date(today, months=-6),as_str=True), get_quarter_ending(add_to_date(today, months=-3))),
+		"last year": lambda: (get_year_start(add_to_date(today, years=-1)), get_year_ending(add_to_date(today, years=-1))),
```

Comment:
last year and next year do not work. A fix is needed in the `get_year_ending` function.

You'll have to add this as the first line, to convert from string to date:
`date = getdate(date)`